src/Workspaces/SharedUtilitiesAndExtensions/Compiler/VisualBasic/Simplification/VisualBasicInferredMemberNameSimplifier.vb

```diff
@@ -32,7 +32,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Simplification
         End Function
 
         Friend Function CanSimplifyNamedFieldInitializer(node As NamedFieldInitializerSyntax) As Boolean
-            If RemovalCausesAmbiguity(DirectCast(node.Parent, ObjectMemberInitializerSyntax).Initializers, node) Then
+            If node.Parent.IsParentKind(SyntaxKind.ObjectCreationExpression) OrElse RemovalCausesAmbiguity(DirectCast(node.Parent, ObjectMemberInitializerSyntax).Initializers, node) Then
```

Comment:
1. should likely have omment explaining this. 
2. what happens with nested initializers?  you can have those in C# like so:

```c#
new Something
{
    C = { Y = y, Z  = z },
}
```

woudl that be an issue for VB here?

Reply:
Yep this is a good point. The previous code was more robust in that it enforced with an `InvalidCastException` that the parent was an `ObjectMemberInitializer`. I've pulled that out as a variable to maintain the old behaviour. That way there'll be no jumping through an unknown intermediate node and no risk of what you've described.

I have also now checked this is correct according to [syntax in the docs](https://docs.microsoft.com/en-us/dotnet/visual-basic/reference/language-specification/expressions#object-creation-expressions) which say any `FieldInitializer` is always part of an `ObjectMemberInitializer`. They also say `ObjectMemberInitializer` is always part of `AnonymousObjectCreationExpression` or `ObjectCreationExpressionInitializer` (they're from 2017 so don't mention named tuples).

I felt a bit thin on information to add to the comment so if there's any questions you have that you feel a comment should explain let me know